src/cmap/auth/providers.ts

```diff
@@ -10,5 +10,12 @@ export const AuthMechanism = Object.freeze({
   MONGODB_X509: 'MONGODB-X509'
 } as const);
 
+/** @public */
+export const $EXTERNAL_AUTH_SOURCE_MECHANISMS = new Set<AuthMechanism | undefined>([
```

Comment:
Since all three auth providers are defined above, they'll never be `undefined`.  We should be able to remove the `| undefined` from the type of the set

Reply:
Also, can we remove the `$` from the name of the variable?

Reply:
Oh I see, because when we use it, we potentially pass in an undefined value.  That's not something I had considered! 
 I'm okay with leaving the undefined then 😄

Reply:
The reason we had the $ was to distinguish these mechanisms related to the `$external` authSource -- not sure if there's a better naming to include that context

Reply:
100% agree. Basically in this case, external does not mean like "outside of MongoDB" -- we're brainstorming other namings :)

Reply:
I second Bailey's suggestion here, is there really a chance of this causing confusion? @nbbeeken